packages/netlify-cms-core/src/valueObjects/EditorComponent.js

```diff
@@ -1,39 +1,35 @@
-import { Record, fromJS } from 'immutable';
+import { fromJS } from 'immutable';
 import { isFunction } from 'lodash';
 
 const catchesNothing = /.^/;
-/* eslint-disable no-unused-vars */
-const EditorComponent = Record({
-  id: null,
-  label: 'unnamed component',
-  icon: 'exclamation-triangle',
-  fields: [],
-  pattern: catchesNothing,
-  fromBlock(match) {
-    return {};
-  },
-  toBlock(attributes) {
-    return 'Plugin';
-  },
-  toPreview(attributes) {
-    return 'Plugin';
-  },
-});
-/* eslint-enable */
```

Comment:
Since this was being overhauled, took the opportunity to remove another Immutable.js dependency.